### DISCUSSION TITLE
concurrency报错

### DISCUSSION THREAD

**Discussion by zhouziyiaichifeiniu** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#63] [Q&A]
遇到了如下报错
def test_content():
        with open("test/tmp3/download_result.fragment", "rb") as download_file:
            download_fragment = pickle.load(download_file)
    
        target_hash = ["45acace8e984465459c893197e593c36daf653db", "3b68110847941b84e8d05417a5b2609122a56314"]
    
        for th in target_hash:
>           assert th in download_fragment, f"download hash mismatch, target: {th}, has: {download_fragment.keys()}"
E           AssertionError: download hash mismatch, target: 3b68110847941b84e8d05417a5b2609122a56314, has: dict_keys(['45acace8e984465459c893197e593c36daf653db'])
E           assert '3b68110847941b84e8d05417a5b2609122a56314' in {'45acace8e984465459c893197e593c36daf653db': b'E\x83\xb1\xe4\xc3\r\xaea\x08\x86\x00\xb4jK\x8e0Cb\xe2\xc8/\xc0\x07+\x02r\x81H\x06\xc2\x9a\xf0_ \xcf\xda\xaa\xd1qX\x0e\xb5b\xab\x16\x91\x03G$\xd2(\xff\x18\x00\t\x90@\x12\x04\xf0\xe8\xba\xb0\x07x\x00\x12\xe6\x00\x1d`\x80\x07V\x80\x07Ba$\x15s1\x19\x138\xbaD\r\xdc\xd0 \x98Q;\x82h\xf62B\'U\x00n\xfaG…………

请问这里是哪一部分写糊了导致的问题？

**Comment by TANSixu** [comment]
Hi，看起来是有一个chunk没有下载完。Concurrency应该会下载两个chunk，从报错信息来看，你保存的下载文件中只有hash为 45acace8e984465459c893197e593c36daf653db 的chunk。可以检查一下是否第二个chunk还没下载完就将结果输出到文件了。

**Comment by jack2223333** [comment]
我也是这个问题，请问你解决了吗
